python/pyarrow/parquet/core.py

```diff
@@ -1327,17 +1327,15 @@ def __init__(self, path_or_paths, filesystem=None, schema=None, *, filters=None,
                     except ValueError:
                         filesystem = LocalFileSystem(use_mmap=memory_map)
                 finfo = filesystem.get_file_info(path_or_paths)
-                if finfo.is_file:
-                    single_file = path_or_paths
```

Comment:
I am not sure why the changes from `single_file` to `buffer_reader` and removing this two lines are needed. Could you provide some feedback on why you decided on this changes?

Reply:
I do not have a preference with the naming but do like to keep things as they are.

If I understand correctly, this case is never met so you have removed it? Could you explain that a bit more?

Reply:
Oh, I think I get it now! 😊 
Do we have an existing test for the BufferReader case and if not, can we add one?

Also, can we then change the comment in line 1316: "# check for single fragment dataset" to something that describes this part of the code correctly, like "check for BufferReader or dataset directory"? (not sure I got this right though).

Reply:
Added in https://github.com/apache/arrow/pull/39438/commits/d92d55740398c812860dec21e3713e5e52254756